src/Middleware/WebSockets/test/UnitTests/WebSocketMiddlewareTests.cs

```diff
@@ -496,6 +498,61 @@ public async Task CloseFromCloseReceived_Success()
         }
     }
 
+    [Fact]
+    public async Task WebSocket_Abort_Interrupts_Pending_ReceiveAsync()
+    {
+        WebSocket serverSocket = null;
+
+        var socketWasAccepted = new ManualResetEventSlim();
+        var socketWasAborted = new ManualResetEventSlim();
+        var firstReceiveOccured = new ManualResetEventSlim();
+
+        await using (var server = KestrelWebSocketHelpers.CreateServer(LoggerFactory, out var port, async context =>
+        {
+            Assert.True(context.WebSockets.IsWebSocketRequest);
+            serverSocket = await context.WebSockets.AcceptWebSocketAsync();
+            socketWasAccepted.Set();
+
+            var serverBuffer = new byte[1024];
+
+            try
+            {
+                while (new[] { WebSocketState.Open, WebSocketState.CloseSent }.Contains(serverSocket.State))
+                {
+                    var response = await serverSocket.ReceiveAsync(serverBuffer, default);
+                    firstReceiveOccured.Set();
+                }
+            }
+            catch (ConnectionAbortedException)
+            {
+                socketWasAborted.Set();
+            }
+        }))
+        {
+            var clientBuffer = new byte[1024];
+
+            using (var client = new ClientWebSocket())
+            {
+                await client.ConnectAsync(new Uri($"ws://127.0.0.1:{port}/"), CancellationToken.None);
+
+                var socketWasAcceptedDidNotTimeout = socketWasAccepted.Wait(10000);
+                Assert.True(socketWasAcceptedDidNotTimeout, "Socket was not accepted within the allotted time.");
+
+                await client.SendAsync(clientBuffer, WebSocketMessageType.Binary, false, default);
+
+                var firstReceiveOccuredDidNotTimeout = firstReceiveOccured.Wait(10000);
+                Assert.True(firstReceiveOccuredDidNotTimeout, "First receive did not occur within the allotted time.");
+
+                await Task.Delay(100); // Give the server receive loop enough time to get back around.
```

Comment:
Avoid arbitrary delays. What happens if you remove this?

Reply:
I added it because of the potential for the loop not to get back around to calling ReceiveAsync before the socket was aborted. I'm tweaking the test a bit to avoid the need to the delay and running the test until it fails to see if there are any potential timing issues without it.